src/MyEventEmitter.js

```diff
@@ -1,14 +1,74 @@
+/* eslint-disable max-len */
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.events = {};
+  }
+
+  on(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+    this.events[eventName].push(listener);
+  }
+
+  once(eventName, listener) {
+    const onceListener = (...args) => {
+      this.off(eventName, onceListener);
+      listener(...args);
+    };
+
+    this.on(eventName, onceListener);
+  }
+
+  off(eventName, listener) {
+    if (!this.events[eventName]) {
+      return;
+    }
+
+    this.events[eventName] = this.events[eventName].filter(
+      (registeredListener) => registeredListener !== listener,
+    );
+  }
+
+  emit(eventName, ...args) {
+    if (!this.events[eventName]) {
```

Comment:
Similar to the previous comment, this check is unnecessary in the 'emit' method. If the event doesn't exist, the forEach loop will not be executed, so there's no need for this check.